bats/keyless-tables.bats

```diff
@@ -0,0 +1,369 @@
+
+    run dolt sql -q "SHOW CREATE TABLE keyless;" -r csv
+    [ $status -eq 0 ]
+    [[ "$output" = "CREATE TABLE \`keyless\` (" ]] || false
+    [[ "$output" = "\`c1\` int," ]] || false
+    [[ "$output" = "\`c2\` int" ]] || false
+    [[ "$output" = ")" ]] || false
+
+    run dolt sql -q "SELECT count(*) FROM keyless;" -r csv
+    [ $status -eq 0 ]
+    [[ "$output" = "4" ]] || false
+}
+
+@test "keyless tables maintain insert order" {
+    run dolt sql -q "SELECT * FROM keyless;" -r csv
+    [ $status -eq 0 ]
+    [[ "$output" = "0,0" ]] || false
+    [[ "$output" = "2,2" ]] || false
+    [[ "$output" = "1,1" ]] || false
+    [[ "$output" = "1,1" ]] || false
+}
+
+@test "delete from keyless" {
+    run dolt sql -q "DELETE FROM keyless WHERE c1 = 2;"
+    [ $status -eq 0 ]
+
+    run dolt sql -q "SELECT * FROM keyless;" -r csv
+    [ $status -eq 0 ]
+    [[ "$output" = "0,0" ]] || false
+    [[ "$output" = "1,1" ]] || false
+    [[ "$output" = "1,1" ]] || false
+}
+
+@test "update keyless" {
+    run dolt sql -q "UPDATE keyless SET c0 = 9 WHERE c0 = 2;"
+    [ $status -eq 0 ]
+
+    run dolt sql -q "SELECT * FROM keyless;" -r csv
+    [ $status -eq 0 ]
+    [[ "$output" = "0,0" ]] || false
+    [[ "$output" = "9,2" ]] || false
+    [[ "$output" = "1,1" ]] || false
+    [[ "$output" = "1,1" ]] || false
+}
+
+@test "keyless table import" {
+    cat <<CSV > data.csv
+c0,c1
+0,0
+2,2
+1,1
+1,1
+CSV
+    dolt table import -c tbl data.csv
+    run dolt sql -q "SELECT count(*) FROM tbl;" -r csv
+    [ $status -eq 0 ]
+    [[ "$output" = "4" ]] || false
+    run dolt sql -q "SELECT * FROM tbl;" -r csv
+    [ $status -eq 0 ]
+    [[ "$output" = "0,0" ]] || false
+    [[ "$output" = "2,2" ]] || false
+    [[ "$output" = "1,1" ]] || false
+    [[ "$output" = "1,1" ]] || false
+}
+
+@test "keyless table update" {
+    cat <<CSV > data.csv
+c0,c1
+0,0
+2,2
+1,1
+1,1
+CSV
+    # update is a pure append
+    dolt table import -u keyless data.csv
+    run dolt sql -q "SELECT count(*) FROM keyless;" -r csv
+    [ $status -eq 0 ]
+    [[ "$output" = "8" ]] || false
+    run dolt sql -q "SELECT * FROM keyless;" -r csv
+    [ $status -eq 0 ]
+    [[ "$output" = "0,0" ]] || false
+    [[ "$output" = "2,2" ]] || false
+    [[ "$output" = "1,1" ]] || false
+    [[ "$output" = "1,1" ]] || false
+    [[ "$output" = "0,0" ]] || false
+    [[ "$output" = "2,2" ]] || false
+    [[ "$output" = "1,1" ]] || false
+    [[ "$output" = "1,1" ]] || false
+}
+
+@test "keyless table replace" {
+    cat <<CSV > data.csv
+c0,c1
+7,7
+8,8
+9,9
+CSV
+    dolt table import -r keyless data.csv
+    run dolt sql -q "SELECT count(*) FROM keyless;" -r csv
+    [ $status -eq 0 ]
+    [[ "$output" = "3" ]] || false
+    run dolt sql -q "SELECT * FROM keyless;" -r csv
+    [ $status -eq 0 ]
+    [[ "$output" = "7,7" ]] || false
+    [[ "$output" = "8,8" ]] || false
+    [[ "$output" = "9,9" ]] || false
+
+}
+
+@test "keyless diff against working set" {
```

Comment:
Would be good to see some additional tests here, like delete * followed by inserting the original values. Does it show a diff? (hopefully not)

Updating one column in the first row. What does the diff have?